src/amp-story-player/amp-story-player-impl.js

```diff
@@ -1160,28 +1145,44 @@ export class AmpStoryPlayer {
 
   /**
    * Returns a promise that makes sure current story gets loaded first before
-   * others.
+   * others. When this method is called with any story other than the current
+   * story, it will not resolve until the current story has finished loading.
    * @param {!StoryDef} story
    * @return {!Promise}
    * @private
    */
   currentStoryPromise_(story) {
     if (this.stories_[this.currentIdx_].storyContentLoaded) {
+      // Return a promise signifying that the current story has loaded.
       return Promise.resolve();
     }
 
     if (story.distance !== 0) {
+      // When the current story has not yet loaded and the given story is not
+      // the current story, return a promise that will resolve only when the
+      // current story has loaded.
       return this.currentStoryLoadDeferred_.promise;
     }
 
-    if (this.currentStoryLoadDeferred_) {
-      // Cancel previous story load promise.
-      this.currentStoryLoadDeferred_.reject(
-        `[${LOG_TYPE_ENUM.DEV}] Cancelling previous story load promise.`
-      );
-    }
+    // Cancel previous story load promise.
+    this.currentStoryLoadDeferred_?.reject(
+      `[${LOG_TYPE_ENUM.DEV}] Cancelling previous story load promise.`
+    );
+
+    // Reset `this.currentStoryLoadDeferred_` and have it resolve only when the
+    // current story has loaded.
```

Comment:
I think all the comments you added (1156/1161+/1167/1172+) reads the code but don't add any additional context (as in, anything that is not obvious when first reading the code, edge cases, etc).